cs2/src/state/player.rs

```diff
@@ -87,69 +89,38 @@ impl TryFrom<CBoneStateData> for BoneStateData {
     }
 }
 
-#[derive(Debug, Clone)]
-pub enum PlayerPawnState {
-    Alive(PlayerPawnInfo),
-    Dead,
-}
+impl State for PlayerPawnInfo {
+    type Parameter = EntityHandle<C_CSPlayerPawn>;
 
-impl State for PlayerPawnState {
-    type Parameter = u32;
-
-    fn create(
-        states: &utils_state::StateRegistry,
-        pawn_entity_index: Self::Parameter,
-    ) -> anyhow::Result<Self> {
+    fn create(states: &StateRegistry, handle: Self::Parameter) -> anyhow::Result<Self> {
         let entities = states.resolve::<EntitySystem>(())?;
-
-        let player_pawn = match entities
-            .get_by_handle::<C_CSPlayerPawn>(&EntityHandle::from_index(pawn_entity_index))?
-        {
-            Some(identity) => identity
-                .entity()?
-                .read_schema()
-                .with_context(|| obfstr!("failed to read player pawn data").to_string())?,
-            None => return Ok(Self::Dead),
+        let Some(player_pawn) = entities.get_by_handle(&handle)? else {
+            anyhow::bail!("entity does not exists")
         };
+        let player_pawn = player_pawn.entity()?.read_schema()?;
 
         let player_health = player_pawn.m_iHealth()?;
-        if player_health <= 0 {
-            return Ok(Self::Dead);
-        }
 
         /* Will be an instance of CSkeletonInstance */
         let game_screen_node = player_pawn
             .m_pGameSceneNode()?
             .cast::<CSkeletonInstance>()
             .read_schema()?;
-        if game_screen_node.m_bDormant()? {
-            return Ok(Self::Dead);
-        }
 
         let controller_handle = player_pawn.m_hController()?;
```

Comment:
It would probably be better to use `player_pawn.m_hOriginalController` which will be valid for dead pawns.